lib/syskit/roby_app/rest_api.rb

```diff
@@ -0,0 +1,181 @@
+
+                def make_configured_deployment_info(d)
+                    type = find_type_from_process_server(
+                        syskit_conf.process_server_config_for(d.process_server_name))
+                    return if !type
+
+                    tasks = d.each_orogen_deployed_task_context_model.map do |deployed_task|
+                        Hash['task_name' => deployed_task.name,
+                             'task_model_name' => deployed_task.task_model.name]
+                    end
+
+                    Hash[id: d.object_id,
+                         created: deployment_manager.created_here?(d),
+                         deployment_name: d.process_name,
+                         tasks: tasks,
+                         on: d.process_server_name,
+                         mappings: d.name_mappings,
+                         type: type]
+                end
+
+                PROCESS_SERVER_TYPES = Hash[
+                    Orocos::RemoteProcesses::Client => 'orocos',
+                    UnmanagedTasksManager => 'unmanaged'
+                ]
+
+                # Returns a string that describes the process server type
+                #
+                # @return [String,nil] the description, or nil if the given
+                #   process server should not be exposed by the REST API
+                def find_type_from_process_server(process_server_config)
+                    PROCESS_SERVER_TYPES[process_server_config.client.class]
+                end
+            end
+
+            get '/deployments/available' do
+                by_deployment = Hash.new
+                roby_app.default_pkgconfig_loader.each_available_deployed_task do |info|
+                    key = [info.deployment_name, info.project_name]
+                    deployment_info = (by_deployment[key] ||= Hash.new)
+                    deployment_info[info.task_name] = info.task_model_name
+                end
+                info = by_deployment.map do |(deployment_name, project_name), tasks|
+                    default_deployment_for =
+                        if /^orogen_default_/.match?(deployment_name)
+                            tasks[deployment_name]
+                        end
+                    default_logger =
+                        if tasks[default_logger_name = "#{deployment_name}_Logger"] == 'logger::Logger'
+                            default_logger_name
+                        end
+
+                    Hash[
+                        name: deployment_name,
+                        project_name: project_name,
+                        tasks: tasks.map { |task_name, task_model_name| Hash['task_name' => task_name, 'task_model_name' => task_model_name] },
+                        default_deployment_for: default_deployment_for,
+                        default_logger: default_logger
+                    ]
+                end
+                { deployments: info }
+            end
+
+            get '/deployments/registered' do
```

Comment:
How about some documentation on these endpoints? :-)

Reply:
Docu ... what ? :P

Reply:
... and done